discord/gameStateMessage.go

```diff
@@ -10,19 +10,45 @@ type GameStateMessage struct {
 	lock    sync.RWMutex
 }
 
+type PrivateStateMessage struct {
+	message *discordgo.Message
+	lock sync.RWMutex
+	idUsernameMap map[string]string
+	printedUsers []string
+	privateChannelID string
+	currentUserID string
+}
+
 func MakeGameStateMessage() GameStateMessage {
 	return GameStateMessage{
 		message: nil,
 		lock:    sync.RWMutex{},
 	}
 }
 
+func MakePrivateStateMessage() PrivateStateMessage {
+	return PrivateStateMessage{
+		message: nil,
+		lock:    sync.RWMutex{},
+		idUsernameMap: make(map[string]string),
+		printedUsers: make([]string, 0),
+		privateChannelID: "000000000000000000", // Default Value
+		currentUserID: "",
+	}
+}
+
 func (gsm *GameStateMessage) Exists() bool {
 	gsm.lock.RLock()
 	defer gsm.lock.RUnlock()
 	return gsm.message != nil
 }
 
+func (psm *PrivateStateMessage) Exists() bool {
```

Comment:
Good locking behavior and conventions being used in this file, good work :)

Reply:
Good to hear haha. I actually just copied your similar methods in this file. I think I understand a bit more about how to use lock methods.